### ISSUE TITLE
TensorBoard link names can become too long based on run labels

### ISSUE THREAD

**Issue by gar1t** [guildai/guildai#29]
Guild indiscriminately uses a run's label in the TB link name. Long labels lead to long file names that aren't necessarily helpful.

Approaches:

- Don't use labels at all (simplifies the problem but loses useful functionality)
- Truncate labels at some max length

**Comment by gar1t** [comment]
As of 0.6.5 Guild no longer includes all flag values in trial labels but only those flags that have changed. This should generally shorten the labels.

It's not clear what the max label length should be for TensorBoard - so closing this until we have a better idea of how to address this. Anyone is free to re-open.

**Comment by dvirginz** [comment]
It seems the labels are created by the run flags, even without manually setting labels.
I have the too-long error:
```
(humanGan) dvir@dvir:~/Documents/repos/desantanglGAN$ guild tensorboard
Preparing runs for TensorBoard
Refreshing flags...
WARNING: cannot import flags from /home/dvir/Documents/repos/desantanglGAN/./train.py: /home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
INFO: [OpenGL.acceleratesupport] No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
'debug_mode'
Traceback (most recent call last):
  File "/home/dvir/anaconda3/envs/humanGan/bin/guild", line 8, in <module>
    sys.exit(main())
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/main_bootstrap.py", line 38, in main
    guild.main.main()
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/main.py", line 32, in main
    _main()
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/main.py", line 38, in _main
    main_cmd.main(standalone_mode=False)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/external/click/core.py", line 759, in __call__
    return self.main(*args, **kwargs)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/external/click/core.py", line 714, in main
    rv = self.invoke(ctx)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/external/click/core.py", line 951, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/external/click/core.py", line 552, in invoke
    return callback(*args, **kwargs)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/click_util.py", line 146, in fn
    return fn0(*(args + (Args(**kw),)))
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/commands/tensorboard.py", line 78, in tensorboard
    tensorboard_impl.main(args)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/commands/tensorboard_impl.py", line 39, in main
    monitor.run_once(exit_on_error=True)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/run_util.py", line 74, in run_once
    self._refresh_logdir(runs)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/run_util.py", line 81, in _refresh_logdir
    path = self._ensure_run(name)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/run_util.py", line 94, in _ensure_run
    util.ensure_dir(path)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/site-packages/guild/util.py", line 65, in ensure_dir
    os.makedirs(d)
  File "/home/dvir/anaconda3/envs/humanGan/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
OSError: [Errno 36] File name too long: '/tmp/guild-tensorboard-kxczw1hy/092bbc42 UnetReconstruction:train 2019-12-31 08:17:59 data_dir=_home_dvir_Documents_repos_desantanglGAN_data_faust_mixamo_faust debug_mode=0 gpu_indecies=1 pool_type=SAGPooling template_mesh=_home_dvir_Documents_repos_desantanglGAN_dfaust_data_registrations_train'

```

**Comment by gar1t** [comment]
Thank you for the report! A fix for this will land in the next pre-release.

**Comment by gar1t** [comment]
In addition to this fix, 0.7 will support a `--run-name-flags` option that lets you list a set of flags to use as the run label. In some cases the default label is too long so this option can be used to show only certain flags - or no flags at all if an empty string is specified.

**Comment by gar1t** [comment]
I'm going to close this as I believe the issue is resolved in 0.7.0. Please re-open if you think otherwise and we'll sort it out. Thanks again for the report!
